File: src/Mapping.cpp
#include "Mapping.h"

Position::Position(int line, int column) : line{line}, column{column} {

}

Mapping::Mapping(Position generated, Position original, int source, int name) : generated{generated},
                                                                                original{original}, source{source},
                                                                                name{name} {

}

File: parcel_sourcemap/src/mapping_line/mapping.rs
pub struct Position {
    pub line: i32,
    pub column: i32,
}

impl Position {
    pub fn new(line: i32, column: i32) -> Self {
        Self { line, column }
    }
}

pub struct Mapping {
    pub original: Position,
    pub generated: Position,
    pub source: i32,
    pub name: i32,
}

impl Mapping {
    pub fn new(generated: Position, original: Position, source: i32, name: i32) -> Self {
        Self {
            original,
            generated,
            source,
            name,
        }
    }
}
